### ISSUE TITLE
Why use malloc.h instead of stdlib.h

### ISSUE THREAD

**Issue by shoccho** [dorjoy03/ungzip#1]
I don't know much about it but why are you using malloc.h instead of stdlib.h . From what I have seen malloc.h non-standard header whereas stdlib.h is a [standard C header]( https://en.cppreference.com/w/c/header) that has malloc.

 Also I couldn't find any use of the non standard methods that malloc.h provides over stdlib.  Other than the small fact that malloc.h includes stdio so you don't have to include it here. If you use stdlib here you would also need stdio.  I am just mentioning this because currently I was not able to compile the code in macOS Ventura ( Arm). I would love to try this out on macos.  Is there anything I am missing here regarding  malloc over stdlib?

**Comment by dorjoy03** [comment]
> I don't know much about it but why are you using malloc.h instead of stdlib.h . From what I have seen malloc.h non-standard header whereas stdlib.h is a [standard C header](https://en.cppreference.com/w/c/header) that has malloc.
> 
> Also I couldn't find any use of the non standard methods that malloc.h provides over stdlib. Other than the small fact that malloc.h includes stdio so you don't have to include it here. If you use stdlib here you would also need stdio. I am just mentioning this because currently I was not able to compile the code in macOS Ventura ( Arm). I would love to try this out on macos. Is there anything I am missing here regarding malloc over stdlib?

Good point. I don't see any reason to use malloc.h header file over stdlib.h which will make the code more portable anyway. Does it compile and run if you just include stdlib.h instead of malloc.h?

**Comment by shoccho** [comment]
After changing malloc.h to stdlib.h and also include the stdio.h for fprintf. I compiles and  runs perfectly on macos (arm) and linux ( compiled using clang 15.0.7  on x86_64 )

**Comment by dorjoy03** [comment]
> After changing malloc.h to stdlib.h and also include the stdio.h for fprintf. I compiles and runs perfectly on macos (arm) and linux ( compiled using clang 15.0.7 on x86_64 )

Great :)

**Comment by dorjoy03** [comment]
Thanks for reporting the issues. These are now fixed in master as part of commit 2063690a1821fb68a4309812cfb1d4f7b4176285 .
